lib/fog/libvirt/models/compute/server.rb

```diff
@@ -257,11 +257,134 @@ def cloud_init_volume_name
         private
         attr_accessor :volumes_path
 
-        # This retrieves the ip address of the mac address
+        # This tests the library version before redefining the address
+        # method for this instance to use a method compatible with
+        # earlier libvirt libraries, or uses the dhcp method from more
+        # recent releases.
+        def addresses(service_arg=service, options={})
+          addresses_method = self.method(:addresses_dhcp)
+          # check if ruby-libvirt was compiled against a new enough version
+          # that can use dhcp_leases, as otherwise it will not provide the
+          # method dhcp_leases on any of the network objects.
+          has_dhcp_leases = true
+          begin
+            service.networks.first.dhcp_leases(self.mac)
+          rescue NoMethodError
+            has_dhcp_leases = false
+          rescue
+            # assume some other odd exception.
+          end
+
+          # if ruby-libvirt not compiled with support, or remote library is
+          # too old (must be newer than 1.2.8), then use old fallback
+          if not has_dhcp_leases or service.client.libversion < 1002008
+            addresses_method = self.method(:addresses_ip_command)
+          end
+
+          # replace current definition for this instance with correct one for
+          # detected libvirt to perform check once for connection
+          (class << self; self; end).class_eval do
+            define_method(:addresses, addresses_method)
+          end
+          addresses(service_arg, options)
+        end
+
+        # This retrieves the ip address of the mac address using ip_command
         # It returns an array of public and private ip addresses
         # Currently only one ip address is returned, but in the future this could be multiple
         # if the server has multiple network interface
-        def addresses(service_arg=service, options={})
+        def addresses_ip_command(service_arg=service, options={})
+          mac=self.mac
+
+          # Aug 24 17:34:41 juno arpwatch: new station 10.247.4.137 52:54:00:88:5a:0a eth0.4
+          # Aug 24 17:37:19 juno arpwatch: changed ethernet address 10.247.4.137 52:54:00:27:33:00 (52:54:00:88:5a:0a) eth0.4
+          # Check if another ip_command string was provided
+          ip_command_global=service_arg.ip_command.nil? ? 'grep $mac /var/log/arpwatch.log|sed -e "s/new station//"|sed -e "s/changed ethernet address//g" |sed -e "s/reused old ethernet //" |tail -1 |cut -d ":" -f 4-| cut -d " " -f 3' : service_arg.ip_command
```

Comment:
Maybe you should just return nil in case `ip_command` is nil?

I wonder where all use cases will have the same path and all logs are the same way. Maybe the case where it is empty should be left to the user of the library to decide?